bin/cml/runner.js

```diff
@@ -270,47 +271,55 @@ const runLocal = async (opts) => {
   RUNNER = proc;
   ({ id: RUNNER_ID } = await cml.runnerByName({ name }));
 
-  if (parseInt(idleTimeout) > 0) {
+  if (idleTimeout > 0) {
     const watcher = setInterval(async () => {
       let idle = RUNNER_JOBS_RUNNING.length === 0;
 
-      try {
-        if (cml.driver === 'github') {
-          const job = await cml.runnerJob({ runnerId: RUNNER_ID });
-
-          if (!job && !idle) {
-            winston.error(
-              `Runner is idle as per the GitHub API but busy as per CML internal state. Resetting jobs. Retrying in ${idleTimeout} seconds...`
-            );
-            winston.warn(`CML GitHub driver response: ${JSON.stringify(job)}`);
-            winston.warn(
-              `CML internal state: ${JSON.stringify(RUNNER_JOBS_RUNNING)}`
-            );
-
-            RUNNER_JOBS_RUNNING = [];
+      if (RUNNER_TIMER >= idleTimeout) {
+        try {
+          if (cml.driver === 'github') {
+            const job = await cml.runnerJob({ runnerId: RUNNER_ID });
+
+            if (!job && !idle) {
+              winston.error(
+                `Runner is idle as per the GitHub API but busy as per CML internal state. Resetting jobs. Retrying in ${idleTimeout} seconds...`
+              );
+              winston.warn(
+                `CML GitHub driver response: ${JSON.stringify(job)}`
+              );
+              winston.warn(
+                `CML internal state: ${JSON.stringify(RUNNER_JOBS_RUNNING)}`
+              );
+
+              RUNNER_JOBS_RUNNING = [];
+            }
+
+            if (job && idle) {
+              winston.error(
+                `Runner is busy as per the GitHub API but idle as per CML internal state. Retrying in ${idleTimeout} seconds...`
+              );
+
+              idle = false;
+            }
           }
+        } catch (err) {
+          winston.error(
+            `Error connecting the SCM: ${err.message}. Will try again in ${idleTimeout} secs`
```

Comment:
```suggestion
            `Error connecting the SCM: ${err.message}. Will try again in 1 seconds`
```

Reply:
NO! we are running the timer every second but we are not entering into the api check until we have consumed the time

```js
if (RUNNER_TIMER >= idleTimeout) {
```

Its something I already noted [here](https://github.com/iterative/cml/pull/1009#issuecomment-1131467852).

Reply:
It because we are setting the timer to 0 again

```js
idle = false;
        }

        if (idle) {
          shutdown({ ...opts, reason: `timeout:${idleTimeout}` });
          clearInterval(watcher);
        } else {
          RUNNER_TIMER = 0;
        }
```

Reply:
The behaviour is like it was but waiting a complete idleTime each time, thing that was not doing completely

Reply:
I must be missing something obvious, but behavior seems pretty similar to me 🤔